unmsmfisi-socialapplication/social_app
# Be add notifications post created

## Description

This pull request introduces a notification system using websockets in the social network application. Notifications will be sent in real-time to users when new posts are created.

## Changes Made

- Implemented a notification system for new posts.
- Added a WebSocket event for post creation.
- Updated the frontend to display notifications.
- Modified the backend to send notifications to connected clients.

## Screenshots (if applicable)

[Include screenshots or gifs demonstrating the new notification system, if applicable.]

## How to Test

1. **Run the Application:**
   - According to `readme.md`

2. **Create a New Post:**
   - In the endpoint POST `/post/create`

3. **Observe Notifications:**
   - Create a websocket connection on: `ws://{domain}/notifications`

## WebSocket Implementation Details

The application uses a publish-subscribe pattern for WebSocket communication. When a new post is created, a notification event is published to all connected clients.

- **WebSocket Event:** `postCreated`
- **Payload:** 
```json
{
	"message": "New post created!",
	"data": {
		"Id": 3,
		"UserId": 1,
		"Title": "Title of created post",
		"Description": "Description of create a post",
		"HasMultimedia": true,
		"Public": true,
		"Multimedia": "xd",
		"InsertionDate": "2023-11-10T00:25:15.69096017Z",
		"UpdateDate": "2023-11-10T00:25:15.690960207Z"
	}
}
```

## Additional Notes

- How additional work, must be add notifications by user (when user is logged).
